### DISCUSSION TITLE
Question: Package does not install on hackage. Why?

### DISCUSSION THREAD

**Discussion by ivanperez-keera** [haskell/cabal#8615] [Q&A]
The following package cannot be installed on hackage:

https://hackage.haskell.org/package/ogma-language-c-1.0.6/reports/1

However, it can be installed from hackage if `alex` and `happy` are in the path (BFNC needs them). The following docker image installs it (indirectly):

```Dockerfile
FROM ubuntu:trusty

RUN apt-get update

RUN apt-get install --yes software-properties-common
RUN add-apt-repository ppa:hvr/ghc
RUN apt-get update

RUN apt-get install --yes ghc-9.0.1 cabal-install-2.4
RUN apt-get install --yes libz-dev

ENV PATH=/opt/ghc/9.0.1/bin:/opt/cabal/2.4/bin:$PWD/.cabal-sandbox/bin:$PATH

RUN cabal update
RUN cabal v1-sandbox init
RUN cabal v1-install alex happy
RUN cabal v1-install ogma-cli
```

Is there something in the BNFC or ogma-** cabal files that should be specified differently?

**Comment by ffaf1** [comment]
Cannot test now, but maybe you can do it using `build-tool-depends`?

Il 22 novembre 2022 alle 12:33 Ivan Perez, PhD ha scritto:

**Comment by gbaz** [comment]
Tested and adding `  build-tool-depends: BNFC:bnfc` to the library stanza seems to do the trick.

Without this change, on a local box and not just hackage, the package fails with v2-build in general, if the bnfc executable is not already in scope.

**Comment by ivanperez-keera** [comment]
Thank you. I will try it.

**Reply by Mikolaj** [reply]
Did it work? If so, could you mark an "answer"?
